src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -235,6 +246,20 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
         this.baseResource = `/course-management/${this.courseId}/${this.exercise.type}-exercises/${this.exercise.id}/`;
     }
 
+    showIfSampleSolutionPresent(newExercise: Exercise) {
+        this.modelingExercise = undefined;
+        this.sampleSolution = undefined;
+        this.sampleSolutionUML = undefined;
```

Comment:
Are these lines necessary here?

Reply:
It is added as a guard to not misbehave if the method is called more then once: first with a `newExercise` with an exercise with a sample solution and then with an exercise without it.

For example, [this assertion](https://github.com/ls1intum/Artemis/pull/4328/files#diff-b28592311188e168657dc9c7234911cb8095db5c5363fee5d28346f60f1116b6R271) would fail without these lines.

Reply:
Thank you for the explanation. Yes, these lines make sense if the method is called multiple times like done in the test, but I don't see the method called multiple times in the normal code. So I think the better solution would be to change this usage into two separate tests and use this method only once everywhere.

Reply:
Yes, it is not written multiple times in the source code however it is called in a `subscribe()` callback so theoretically it can be triggered multiple times (although maybe unlikely in the common use case).
[`loadExercise()`](https://github.com/ls1intum/Artemis/blob/feature/modeling-exercises/present-sample-solution/src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts#L189) method indirectly calling this method also uses the same pattern for the `exercise` property.

I can add some comments explaining the purpose of these lines, what do you think?

Reply:
ok, thank you for the explanation. Yes, please add a comment that contains this explanation.

Reply:
Commented the code with the reasoning.